luarules/gadgets/cmd_transport_unit_gadget.lua

```diff
@@ -122,6 +129,36 @@ function gadget:UnitUnloaded(unitID, unitDefID, teamID, transportID)
 	loadedUnits[unitID] = nil
 end
 
+function gadget:RecvLuaMsg(msg, playerID)
+	-- Check if it's our message type
+	if msg:sub(1, 4) == "POS|" then
+		local _, unitID, x, y, z = msg:match("([^|]+)|([^|]+)|([^|]+)|([^|]+)|([^|]+)")
+		unitID = tonumber(unitID)
+		x, y, z = tonumber(x), tonumber(y), tonumber(z)
```

Comment:
this looks like it should be a new command, `CMD_POS` with arguments x, y and z (with the unitID arg being there implicitly since the command has to be given to some unit). similarly `CMD_TSTP` below (ideally with better names in both cases)

Reply:
> the only way is to set a move/fight/patrol command OR change the unit goal in a gadget (<what we are doing here) OR modify the engine

Yes, the way to tell the gadget to change the unit goal is to add a command though.